src/AltinnCore/Common/Services/Implementation/SourceControlSI.cs

```diff
@@ -559,5 +559,47 @@ public HttpResponseMessage CheckoutLatestCommitForSpecificFile(string owner, str
                 return new HttpResponseMessage(HttpStatusCode.InternalServerError);
             }
         }
+
+        /// <summary>
+        /// Stages a specific file changed in working repository.
+        /// </summary>
+        /// <param name="owner">The owner of the repository.</param>
+        /// <param name="repository">The name of the repository.</param>
+        /// <param name="fileName">the entire file path with filen name</param>
+        /// <returns>Http response message as ok if checkout operation is successful.</returns>
+        public HttpResponseMessage StageChange(string owner, string repository, string fileName)
+        {
+            HttpResponseMessage responseMessage = new HttpResponseMessage();
+            try
+            {
+                if (string.IsNullOrEmpty(owner) || string.IsNullOrEmpty(repository) || string.IsNullOrEmpty(fileName))
+                {
+                    HttpResponseMessage badRequest = new HttpResponseMessage(HttpStatusCode.BadRequest);
+                    badRequest.ReasonPhrase = "One or all of the input parameters are null";
+                    return badRequest;
+                }
+                
+                string localServiceRepoFolder = _settings.GetServicePath(owner, repository, AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext));
+                using (Repository repo = new Repository(localServiceRepoFolder))
+                {
+                    FileStatus fileStatus = repo.RetrieveStatus().SingleOrDefault(file => file.FilePath == fileName).State;
+
+                    if (fileStatus == FileStatus.ModifiedInWorkdir || fileStatus == FileStatus.NewInWorkdir)
+                    {
+                        Commands.Stage(repo, fileName);
+                    }
+                }
+
+                // To do return custom reposnse message when a file is not found or when a file with no change is sent to stage
```

Comment:
Is this a TODO? Maybe write it as TODO insted of to do, also maybe document the todo in an issue?

typo: reposnse

Reply:
Agree, It's a overall issue of exception handling that would cover this. I'm not creating a new issue as we already have an issue on that
I have to cancel this PR as @matsgm has already merged my changes with his branch